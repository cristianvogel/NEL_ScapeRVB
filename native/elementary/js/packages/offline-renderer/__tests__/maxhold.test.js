import OfflineRenderer from '../index';
import { el } from '@elemaudio/core';


test('maxhold basics', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 1,
    numOutputChannels: 1,
  });

  // Graph
  core.render(el.maxhold(el.in({channel: 0}), 0));

  // Ten blocks of data
  let inps = [new Float32Array(512 * 10)];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Now we use a very calculated input to step the sparseq
  inps = [Float32Array.from([1, 2, 3, 4, 3, 2, 1])];
  outs = [new Float32Array(inps[0].length)];

  // Drive our sparseq
  core.process(inps, outs);

  expect(outs[0]).toMatchSnapshot();
});

test('maxhold hold time', async function() {
  let core = new OfflineRenderer();

  await core.initialize({
    numInputChannels: 1,
    numOutputChannels: 1,
  });

  // Graph
  core.render(el.maxhold({hold: 1}, el.in({channel: 0}), 0));

  // Ten blocks of data
  let inps = [new Float32Array(512 * 10)];
  let outs = [new Float32Array(512 * 10)];

  // Get past the fade-in
  core.process(inps, outs);

  // Now we use a very calculated input to step the sparseq
  inps = [Float32Array.from([
    1, 2, 3, 4, 3, 2, 1, 1,
    1, 1, 1, 1, 1, 1, 1, 1,
    1, 1, 1, 1, 1, 1, 1, 1,
    1, 1, 1, 1, 1, 1, 1, 1,
    1, 1, 1, 1, 1, 1, 1, 1,
    1, 1, 1, 1, 1, 1, 1, 1,
  ])];
  outs = [new Float32Array(inps[0].length)];

  // Drive our sparseq
  core.process(inps, outs);

  expect(outs[0]).toMatchSnapshot();
});
