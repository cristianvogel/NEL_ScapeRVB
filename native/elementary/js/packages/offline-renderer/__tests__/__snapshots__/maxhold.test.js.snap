// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maxhold basics 1`] = `
Float32Array [
  1,
  2,
  3,
  4,
  4,
  4,
  4,
]
`;

exports[`maxhold hold time 1`] = `
Float32Array [
  1,
  2,
  3,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  4,
  1,
]
`;
